File: leetcode/206-ReverseLinkedList/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        curr, prev = head, None
        while curr:
            curr.next, prev, curr  = prev, curr, curr.next
        return prev
        

File: leetcode/206-ReverseLinkedList/solution.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function (head) {
  let pre = null;
  let curr = head;

  while (curr) {
    nextTemp = curr.next;
    curr.next = pre;
    pre = curr;
    curr = nextTemp;
  }
  return pre;
};